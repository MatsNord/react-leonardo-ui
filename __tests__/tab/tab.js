import React from 'react';
import renderer from 'react-test-renderer';

import Tab from '../../src/tab';
import Tabset from '../../src/tabset';

test('Empty tabset renders correctly', () => {
  const tree = renderer.create(
    <Tabset />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Empty tab renders correctly', () => {
  const tree = renderer.create(
    <Tab />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Tabset with simple items renders correctly', () => {
  const tree = renderer.create(
    <Tabset>
      <Tab>Item 1</Tab>
      <Tab>Item 2</Tab>
    </Tabset>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Tabset with complex items renders correctly', () => {
  const tree = renderer.create(
    <Tabset>
      <Tab>
        <Tab.Aside>Aside 1</Tab.Aside>
        <Tab.Text>Text 1</Tab.Text>
      </Tab>
      <Tab>
        <Tab.Text>Text 2</Tab.Text>
        <Tab.Aside>Aside 2</Tab.Aside>
      </Tab>
    </Tabset>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Tabset with modifiers renders correctly', () => {
  const tree = renderer.create(
    <Tabset fill variant="inverse" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Tab with states renders correctly', () => {
  const tree = renderer.create(
    <Tab active disabled />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Tab with modifiers renders correctly', () => {
  const tree = renderer.create(
    <Tab variant="inverse" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Tabset passes props correctly', () => {
  const onClick = jest.fn(() => {});

  const tree = renderer.create(
    <Tabset className="my-class" data-x="x">
      <Tab data-x="x" title="Item title" onClick={onClick}>Item 1</Tab>
    </Tabset>
  ).toJSON();
  expect(tree).toMatchSnapshot();

  tree.children[0].props.onClick();
  expect(onClick.mock.calls.length).toBe(1);
});
