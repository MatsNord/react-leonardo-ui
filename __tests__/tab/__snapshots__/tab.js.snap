exports[`test Empty tab renders correctly 1`] = `
<li
  className="lui-tab" />
`;

exports[`test Empty tabset renders correctly 1`] = `
<ul
  className="lui-tabset" />
`;

exports[`test Tab with modifiers renders correctly 1`] = `
<li
  className="lui-tab lui-tab--inverse" />
`;

exports[`test Tab with states renders correctly 1`] = `
<li
  className="lui-tab lui-active lui-disabled" />
`;

exports[`test Tabset passes props correctly 1`] = `
<ul
  className="lui-tabset my-class"
  data-x="x">
  <li
    className="lui-tab"
    data-x="x"
    onClick={[Function]}
    title="Item title">
    Item 1
  </li>
</ul>
`;

exports[`test Tabset with complex items renders correctly 1`] = `
<ul
  className="lui-tabset">
  <li
    className="lui-tab">
    <div
      className="lui-tab__aside">
      Aside 1
    </div>
    <div
      className="lui-tab__text">
      Text 1
    </div>
  </li>
  <li
    className="lui-tab">
    <div
      className="lui-tab__text">
      Text 2
    </div>
    <div
      className="lui-tab__aside">
      Aside 2
    </div>
  </li>
</ul>
`;

exports[`test Tabset with modifiers renders correctly 1`] = `
<ul
  className="lui-tabset lui-tabset--fill lui-tabset--inverse" />
`;

exports[`test Tabset with simple items renders correctly 1`] = `
<ul
  className="lui-tabset">
  <li
    className="lui-tab">
    Item 1
  </li>
  <li
    className="lui-tab">
    Item 2
  </li>
</ul>
`;
